association_backend/association/domain/entities/subscription_entities.py

```diff
@@ -0,0 +1,76 @@
+from dataclasses import dataclass
+from datetime import datetime
+from enum import Enum
+from typing import Optional
+from zoneinfo import ZoneInfo
+
+from dateutil.relativedelta import relativedelta
+from sqlalchemy import Column, DateTime, String, Table
+from sqlalchemy.orm import registry
+
+rome_tz = ZoneInfo("Europe/Rome")
+
+
+@dataclass
+class UserData:
+    email: str
+    user_id: str
+
+
+class SubscriptionState(str, Enum):
+    PENDING = "pending"
+    ACTIVE = "active"
+    EXPIRED = "expired"
+
+    def __str__(self) -> str:
+        return str.__str__(self)
+
+
+@dataclass
+class Subscription:
+    user_id: str
+    creation_date: datetime
+    stripe_session_id: str
+    state: SubscriptionState
+    payment_date: Optional[datetime] = None
+    stripe_id: Optional[str] = ""
+    stripe_customer_id: Optional[str] = ""
+    expiration_date: Optional[datetime] = None
+
+    def is_payed(self):
+        return self.payment_date is not None
+
+    def is_expired(self):
+        return (
+            self.payment_date
+            and self.payment_date < datetime.now(rome_tz) - relativedelta(years=1)
+            or False
+        )
+
+    def get_calculated_state(self) -> SubscriptionState:
+        if not self.payment_date:
+            return SubscriptionState.PENDING
+        elif self.payment_date < datetime.now(rome_tz) - relativedelta(years=1):
+            return SubscriptionState.EXPIRED
+        else:
+            return SubscriptionState.ACTIVE
```

Comment:
I would avoid having calculated statuses, the status should be a field on the model and should change via cron or something similar

Reply:
I agree with you, this method will be used by the cron to set the `state` field

This is WIP because expired is based on obj.payment_date

Reply:
I removed this method
Now you can see the logic of the status update in the service `handle_customer_subscription_updated.py`